### PR TITLE
Persist shortlist filter (version 2)

### PR DESCRIPTION
**Pull request by novotnyjakub**
### Context

This replaces the closed PR #97 with changes applied from `master` related to location search.

Every filter action on the shortlisting page is persisted to the `Shortlist` model so that the Matchmaker can come back to it at any time and continue.

### Changes proposed in this pull request

* Change the shortlisting mechanics from `show` into `edit+update` actions
* Extract the foster parents list into a query object
* Repurpose the filter to extract and store the filter from/to the persisted `Shortlist`

### Guidance to review

After changing the filter criteria, going back to the dashboard and then again to the shortlist, the filter should be persisted exactly as left off.

### CODE REVIEW

File: app/queries/shortlist_foster_parent_query.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    @latitude = latitude
+    @longitude = longitude
+    @search_radius = search_radius
+  end
+
+  def call
+    FosterParent
+      .joins(:placement_suitability)
+      .includes(:placement_suitability)
+      .yield_self(&method(:placements_part))
```
Comment (goodviber) [inline app/queries/shortlist_foster_parent_query.rb:13]: could this also be written as `yield_self(&placements_part)` ?

Comment (novotnyjakub) [reply app/queries/shortlist_foster_parent_query.rb:13]: No, the method expects a block and this is the only way to "convert" methods into blocks. To allow for the syntax you're writing, we'd need to define something like `placements_patrt = Proc.new {}` but that would loose the point of readability within the class. 

I tried to see if that can be achieved, but this is the only way I found working 🤔

Comment (goodviber) [reply app/queries/shortlist_foster_parent_query.rb:13]: Yes , that makes sense

**Review by goodviber — APPROVED** [review]
LGTM - maybe the call could look for the PlacementSuitability first, then pluck the FosterParent, but not sure that would be more efficient

**Comment by novotnyjakub** [comment]
I'm not sure what performance difference would it make, it's certainly something we can look into later. I'd say it has slightly different semantics if we return `PlacementSuitability` collection here as the main objective of the query is to return `FosterParent` collection. But again, these are not committing decisions that can't easily be changed in this single place with test coverage.

**Pull request merged**